Pinta.Tools/Tools/GradientTool.cs

```diff
@@ -25,33 +25,50 @@
 // THE SOFTWARE.
 
 using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
 using Cairo;
+using Gtk;
 using Pinta.Core;
 
 namespace Pinta.Tools;
 
 public sealed class GradientTool : BaseTool
 {
 	private readonly IPaletteService palette;
-	PointD startpoint;
-	bool tracking;
+
 	private ImageSurface? undo_surface;
-	MouseButton button;
+	private GradientData? undo_data;
+
+	private bool is_newly_created = false;
+
+	//The 'button' variable was split into two so handles can be moved with a different button to the one used when creating the gradient that defines its colors.
+	public MouseButton color_button;
```

Comment:
Would it be more clear to just store whether the gradient should be reversed, rather than the mouse button that was originally used?